engines/bops_core/app/services/bops_core/sgid_authentication_service.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module BopsCore
+  class SgidAuthenticationService
+    attr_reader :sgid, :purpose
+
+    def initialize(sgid, purpose: "magic_link")
+      @sgid = sgid
+      @purpose = purpose
+    end
+
+    def locate_resource
+      GlobalID::Locator.locate_signed(sgid, for: purpose)
+    end
+
+    def expired_resource
+      gid = parse_global_id
+      return nil unless gid
+
+      gid.model_class.find_by(id: gid.model_id)
+    rescue ActiveRecord::RecordNotFound
+      nil
+    end
```

Comment:
Do we ever use the return value of this, or is it effectively just a boolean? calling it `resource_expired?` or something and making it true or false might be nicer if we only ever care that the resource exists.